### PR TITLE
Edit/Create Projects on page in a popup

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
https://trello.com/c/djoHeMob

Requires https://github.com/openshift/origin-web-common/pull/135

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
I have not removed the create project and/or edit project pages. Not sure if we want them removed as part of this PR or not.

File: app/views/projects.html [hunk 1]
```diff
@@ -25,9 +25,13 @@ <h2 class="text-center">Loading...</h2>
                         <h1>My Projects</h1>
                         <div class="projects-options">
                           <div class="projects-add" ng-if="canCreate">
-                            <a href="create-project" class="btn btn-md btn-primary">
+                            <button ng-click="createProject()" class="btn btn-md btn-primary">
+                              <span class="fa fa-plus"></span>
                               Create Project
-                            </a>
+                            </button>
+                            <origin-modal-popup shown="showNewProjectPanel" modal-title="Create Project" on-close="closeNewProjectPanel()">
```
Comment (spadgett) [inline app/views/projects.html]: As much as possible, I'd like to be consistent in how we handle callbacks for our components. It looks like `shown` and `on-close` are handled differently here (as well as `on-cancel` for `create-project`).

File: app/views/projects.html [hunk 2]
```diff
@@ -101,7 +105,7 @@ <h2 class="h1">
                                 </a>
                               </li>
                               <li role="menuitem">
-                                <a ng-href="project/{{project.metadata.name}}/edit?then=./">
+                                <a ng-href="#" ng-click="editProject(project)">
```
Comment (spadgett) [inline app/views/projects.html]:
```
<a href="" ng-click="editProject(project)">
```

File: app/scripts/controllers/projects.js [hunk 3]
```diff
@@ -109,6 +109,35 @@ angular.module('openshiftConsole')
...
+    };
+
+    $scope.onNewProject = function() {
+      $scope.showNewProjectPanel = false;
+    };
+
+    $scope.showEditProjectPanel = false;
+
+    $scope.editProject = function(project) {
+      $scope.edittingProject = project;
```
Comment (spadgett) [inline app/scripts/controllers/projects.js]: `editingProject`

File: app/views/projects.html (hunk 1 above)
Comment (jeff-phillips-18) [inline app/views/projects.html]: I could change either of the components but that would need to be done there and not here. Most of the components bind functions using < or =, I used & for the new origin-modal-popup component. Should I change that?

File: app/views/projects.html (hunk 2 above)
Comment (jeff-phillips-18) [inline app/views/projects.html]: fixed

File: app/scripts/controllers/projects.js (hunk 3 above)
Comment (jeff-phillips-18) [inline app/scripts/controllers/projects.js]: fixed

File: app/views/projects.html (hunk 1 above)
Comment (spadgett) [inline app/views/projects.html]: Right, understood. I suggest we should use `<` or `=` like the other components. I'd at least like to be consistent with callbacks inside the same component. Here `shown` is `<` and `onClose` is `&` in the same component.

Sorry, I missed this reviewing https://github.com/openshift/origin-web-common/pull/135

(As an aside, I find callbacks in Angular components generally weird and confusing...)

Comment (spadgett) [inline app/views/projects.html]: Wondering if `onShown` is not better as well for consistent naming.

Comment (jeff-phillips-18) [inline app/views/projects.html]: shown is a flag not a function.

Comment (jeff-phillips-18) [inline app/views/projects.html]: I can rename the controller variable to reduce confusion.

Comment (spadgett) [inline app/views/projects.html]: Ah OK please disregard then :)

Comment (jeff-phillips-18) [inline app/views/projects.html]: Fixed and renamed the variables.

**Comment by spadgett** [comment]
> I have not removed the create project and/or edit project pages. Not sure if we want them removed as part of this PR or not.

If we remove them, we might think about redirects for the old URLs (maybe just /create-project) in case someone has a link / bookmark. Not too worried about it for this PR.

File: app/views/projects.html [hunk 4]
```diff
@@ -25,9 +25,13 @@ <h2 class="text-center">Loading...</h2>
                         <h1>My Projects</h1>
                         <div class="projects-options">
                           <div class="projects-add" ng-if="canCreate">
-                            <a href="create-project" class="btn btn-md btn-primary">
+                            <button ng-click="createProject()" class="btn btn-md btn-primary">
+                              <span class="fa fa-plus"></span>
```
Comment (spadgett) [inline app/views/projects.html]: `aria-hidden="true"`

File: app/views/projects.html [hunk 5]
```diff
@@ -101,7 +105,7 @@ <h2 class="h1">
                                 </a>
                               </li>
                               <li role="menuitem">
-                                <a ng-href="project/{{project.metadata.name}}/edit?then=./">
+                                <a ng-href="" ng-click="editProject(project)">
```
Comment (spadgett) [inline app/views/projects.html]: nit: `href=""`

**Comment by spadgett** [comment]
If we remove the create project page, we'll also need to update our e2e tests.

https://github.com/spadgett/origin-web-console/blob/f6611831bc6b9ba10b0b7a7c8b90e66e7ddc940b/test/integration/page-objects/createProject.js

**Comment by jeff-phillips-18** [comment]
Created issue #1917 to remove the pages.

**Comment by jeff-phillips-18** [comment]
Removed WIP status now that we have updated the version of origin-web-common.
Ready for final reviews/merge.

**Comment by spadgett** [comment]
Hey @jeff-phillips-18. I pulled the changes down, and everything works well. Thanks.

A couple of small style questions for @openshift/team-ux-review. I won't block the PR on any of these, particularly if they require origin-web-common changes. Just small things I noticed.

1. Should there be any margin between the `+` and the text on the button? I'm not sure if there is a standard style for this in Patternfly.

No margin (current):

openshift web console 2017-08-08 08-15-56

2px margin:

openshift web console 2017-08-08 08-17-48

2. Should we remove `input-lg` and `btn-lg` styles now that the form is only in a dialog?

`input-lg` (current):

openshift web console 2017-08-08 08-18-59

Without `input-lg`:

openshift web console 2017-08-08 08-20-42

3. I noticed the dialog is spaced farther from the kebab than the actions dropdown and feels a little too far away. I assume the spacing should be about the same.

**Comment by spadgett** [comment]
Added Trello card https://trello.com/c/djoHeMob for QE testing.

**Comment by spadgett** [comment]
[test] to make sure this doesn't break our e2e tests

**Comment by spadgett** [comment]
It looks like we do have padding on the button in the landing page, so we should be consistent here so they look the same.

screen shot 2017-08-08 at 9 54 35 am

File: app/views/projects.html (hunk 4 above)
Comment (jeff-phillips-18) [inline app/views/projects.html]: Fixed

File: app/views/projects.html (hunk 5 above)
Comment (jeff-phillips-18) [inline app/views/projects.html]: Fixed

**Comment by jeff-phillips-18** [comment]
Fixed the create button spacing.

Added issue for input and button sizes: https://github.com/openshift/origin-web-common/issues/142

File: app/styles/_buttons.less [hunk 6]
```diff
@@ -41,3 +41,7 @@
     text-decoration: none;
   }
 }
+
+.icon-button-text {
+  padding-left: 4px;
```
Comment (spadgett) [inline app/styles/_buttons.less]: 👍  thanks for making this a generic style

File: app/views/projects.html [hunk 7]
```diff
@@ -101,7 +105,7 @@ <h2 class="h1">
                                 </a>
                               </li>
                               <li role="menuitem">
-                                <a ng-href="project/{{project.metadata.name}}/edit?then=./">
+                                <a g-href="" ng-click="editProject(project)">
```
Comment (spadgett) [inline app/views/projects.html]: typo: `g-href`

Comment (jeff-phillips-18) [inline app/views/projects.html]: ugh

Comment (jeff-phillips-18) [inline app/views/projects.html]: Fixed

**Comment by jeff-phillips-18** [comment]
Fixed popup vertical positioning for projects with descriptions.

**Comment by spadgett** [comment]
[merge]

**Pull request merged**